languageserver/integration/commands.go

```diff
@@ -208,14 +210,70 @@ func (i *FlowIntegration) sendTransaction(conn protocol.Conn, args ...interface{
 		return nil, fmt.Errorf("failed to parse JSON arguments")
 	}
 
-	signer, err := i.state.EmulatorServiceAccount()
+	serviceAccount, err := i.state.EmulatorServiceAccount()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get service account, err: %w", err)
 	}
-	signer.SetAddress(flow.HexToAddress(signers[0]))
 
-	const gasLimit uint64 = 1000
-	_, txResult, err := i.sharedServices.Transactions.Send(signer, code, "", gasLimit, txArgs, "")
+	payer := serviceAccount.Address()
+	keyIndex := serviceAccount.Key().Index()
+
+	// We need to check if service account among authorizers
+	hasServiceAccount := false
+
+	signerAccounts := make([]flowkit.Account, len(signers))
+	authorizers := make([]flow.Address, len(signers))
+	for i, v := range signers {
+		address := flow.HexToAddress(v)
+
+		// Clone service account
+		account := *serviceAccount
+		account.SetAddress(address)
+
+		signerAccounts[i] = account
+		authorizers[i] = address
+
+		if address == payer {
+			hasServiceAccount = true
+		}
+	}
+
+	// If serviceAccount is not in signers list, we will add it to handle payer role properly
+	if !hasServiceAccount {
+		signerAccounts = append(signerAccounts, *serviceAccount)
+	}
+
+	tx, err := i.sharedServices.Transactions.Build(
+		payer,
+		authorizers,
+		payer,
+		keyIndex,
+		code,
+		"",
+		MaxGasLimit,
+		txArgs,
+		"",
+	)
+
+	if err != nil {
+		return nil, errorWithMessage(conn, ErrorMessageTransactionError, err)
+	}
+
+	for _, signer := range signerAccounts {
+		err = tx.SetSigner(&signer)
+		if err != nil {
+			return nil, err
+		}
+
+		tx, err = tx.Sign()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	txBytes := []byte(fmt.Sprintf("%x", tx.FlowTransaction().Encode()))
```

Comment:
Why can't it just be `tx.FlowTransaction().Encode()`?

Reply:
Got this bit from Bjarte without actually looking at return types :sweat_smile:

Reply:
Actually, it doesn't work without this conversion:

```go
transaction error: &errors.errorString{s:"failed to decode partial transaction from \xf9\x01k\xf8ٸx/// pragma arguments (5)\n\ntransaction(number: Int) {\n\tprepare(signer: AuthAccount){\n\t\t// log(amount)\n\t\tlog(number)\n\t}\n}\nܛ{\"type\":\"Int\",\"value\":\"5\"}\n\xa0\x84o\x02cL\xb5\x1b\x0e\xff\x9c\x8a\xc3\xca\xc52\xed.\xf7\x95\x01_\x9f\xef\x11\xa4\u007f\xf0\xd9\xf1\xb5|y\x82'\x0f\x88\xf8\xd6\xe0Xk\n ǀ\t\x88\xf8\xd6\xe0Xk\n \xc7Ɉ\x01\xcf\x0e//qTP\xf8F\xf8D\x01\x80\xb8@a&av\xb1^d\xbdi\xff\x10\x95\x87\xfa\x19\x84e\xc8\x1c\xd4\n0֮Z\xc6\xdcԹ\xf1\x9duq\x192xU+\xa1\xb0Yj\xaaaM\xd3\xd8\xcb+#\x93<og\x04\xa1+\x9b\xaf4\xc7 8a\xf8F\xf8D\x80\x80\xb8@\xe3\xbc\xd3D[7\xf6\xf6\xa8\xe4\xc4\xccӕ\xca\",\xbb\xc7,\xf17\x1c\xec\\\xa3:\xeexT%/\x99'>L\rн\xa7\x95\xf2\x86\xacd\xddQ\x8f┵\xbf̘\xfc@\xb6\x14\xa74\x89\xfdSu: encoding/hex: invalid byte: U+00F9 'ù'"}
```